In [232]:
import pandas as pd
import numpy as np
from geopy.geocoders import Nominatim 
import datetime
from sklearn import preprocessing

In [254]:
poke=pd.read_csv("/Users/Shuxian/Documents/CSYE7200-Fall2016/predictemall/300k.csv",low_memory=False,header=0)

In [255]:
poke.head(5)

pokemonId   latitude   longitude    appearedLocalTime  \
0         16  20.525745  -97.460829  2016-09-08T03:57:45   
1        133  20.523695  -97.461167  2016-09-08T03:57:37   
2         16  38.903590  -77.199780  2016-09-08T03:57:25   
3         13  47.665903 -122.312561  2016-09-08T03:56:22   
4        133  47.666454 -122.311628  2016-09-08T03:56:08   

                            _id           cellId_90m          cellId_180m  \
0  NTgxMDkzOTk4MTM5MjUwMjIzNw==  9645139108510564000  9645139108711890000   
1  OTQ1NDgzODc1MjM3NDEzMTI2MQ==  9645139109852742000  9645139109785633000   
2  NTQ0OTQ0NDA1Nzg2ODg3OTg2OQ==  9923201472785285000  9923201472986612000   
3  NTU2MTU1NDM4NzA2MDk1MDcxNw==  6093392705025474600  6093392705092583400   
4  MTY2ODg4MTAzMTczMDE0MTUwNTM=  6093392707709829100  6093392707776938000   

           cellId_370m          cellId_730m         cellId_1460m  ...    \
0  9645139108443455000  9645139109517197000  9645139113812165000  ...     
1  9645139110590940000  9645139109517197000  9645139113812165000  ...     
2  9923201473791918000  9923201477013144000  9923201481308110000  ...     
3  6093392705897889800  6093392702676664300  6093392715561566200  ...     
4  6093392708045373400  6093392711266598900  6093392715561566200  ...     

  cooc_143 cooc_144 cooc_145  cooc_146  cooc_147 cooc_148  cooc_149  cooc_150  \
0    False    False    False     False     False    False     False     False   
1    False    False    False     False     False    False     False     False   
2    False    False    False     False     False    False     False     False   
3    False    False    False     False     False    False     False     False   
4    False    False    False     False     False    False     False     False   

   cooc_151  class  
0     False     16  
1     False    133  
2     False     16  
3     False     13  
4     False    133  

[5 rows x 208 columns]

#### Drop pokemonId, _id, weatherIcon and S2 columns

poke.columns.get_loc('weatherIcon')

In [256]:
poke.drop(poke.columns[[0,3,4,5,6,7,8,9,10,11,28]],axis=1,inplace=True)

In [257]:
poke.head(5)

latitude   longitude appearedTimeOfDay  appearedHour  appearedMinute  \
0  20.525745  -97.460829             night             5              57   
1  20.523695  -97.461167             night             5              57   
2  38.903590  -77.199780             night             5              57   
3  47.665903 -122.312561             night             5              56   
4  47.666454 -122.311628             night             5              56   

  appearedDayOfWeek  appearedDay  appearedMonth  appearedYear  terrainType  \
0         dummy_day            8              8          2016           14   
1         dummy_day            8              8          2016           14   
2         dummy_day            8              8          2016           13   
3         dummy_day            8              8          2016            0   
4         dummy_day            8              8          2016            0   

   ...   cooc_143 cooc_144 cooc_145 cooc_146  cooc_147  cooc_148  cooc_149  \
0  ...      False    False    False    False     False     False     False   
1  ...      False    False    False    False     False     False     False   
2  ...      False    False    False    False     False     False     False   
3  ...      False    False    False    False     False     False     False   
4  ...      False    False    False    False     False     False     False   

   cooc_150  cooc_151  class  
0     False     False     16  
1     False     False    133  
2     False     False     16  
3     False     False     13  
4     False     False    133  

[5 rows x 197 columns]

In [237]:
poke.appearedYear.unique()

array([2016])

In [238]:
poke.appearedMonth.unique()

array([8])

#### Mon-Sun===>1~7

In [258]:
# poke[poke.columns[[8,7,6]]]
poke['DateTime'] = poke[poke.columns[[8,7,6]]].apply(lambda s : datetime.datetime(*s),axis = 1)

In [259]:
poke['appearedDayOfWeek']=poke['DateTime'].dt.dayofweek
poke['appearedDayOfWeek']=poke['appearedDayOfWeek']+1

In [260]:
poke.head(5)

latitude   longitude appearedTimeOfDay  appearedHour  appearedMinute  \
0  20.525745  -97.460829             night             5              57   
1  20.523695  -97.461167             night             5              57   
2  38.903590  -77.199780             night             5              57   
3  47.665903 -122.312561             night             5              56   
4  47.666454 -122.311628             night             5              56   

   appearedDayOfWeek  appearedDay  appearedMonth  appearedYear  terrainType  \
0                  1            8              8          2016           14   
1                  1            8              8          2016           14   
2                  1            8              8          2016           13   
3                  1            8              8          2016            0   
4                  1            8              8          2016            0   

     ...     cooc_144 cooc_145 cooc_146 cooc_147  cooc_148  cooc_149  \
0    ...        False    False    False    False     False     False   
1    ...        False    False    False    False     False     False   
2    ...        False    False    False    False     False     False   
3    ...        False    False    False    False     False     False   
4    ...        False    False    False    False     False     False   

   cooc_150  cooc_151  class   DateTime  
0     False     False     16 2016-08-08  
1     False     False    133 2016-08-08  
2     False     False     16 2016-08-08  
3     False     False     13 2016-08-08  
4     False     False    133 2016-08-08  

[5 rows x 198 columns]

In [261]:
poke.appearedTimeOfDay.unique()

array(['night', 'evening', 'afternoon', 'morning'], dtype=object)

#### ['afternoon','evening','morning', 'night']==>[1,2,3,4]

In [262]:
le = preprocessing.LabelEncoder()
le.fit(poke.appearedTimeOfDay.unique())
list(le.classes_)
poke['appearedTimeOfDay']=le.transform(poke['appearedTimeOfDay'])+1

In [263]:
poke.head(5)

latitude   longitude  appearedTimeOfDay  appearedHour  appearedMinute  \
0  20.525745  -97.460829                  4             5              57   
1  20.523695  -97.461167                  4             5              57   
2  38.903590  -77.199780                  4             5              57   
3  47.665903 -122.312561                  4             5              56   
4  47.666454 -122.311628                  4             5              56   

   appearedDayOfWeek  appearedDay  appearedMonth  appearedYear  terrainType  \
0                  1            8              8          2016           14   
1                  1            8              8          2016           14   
2                  1            8              8          2016           13   
3                  1            8              8          2016            0   
4                  1            8              8          2016            0   

     ...     cooc_144 cooc_145 cooc_146 cooc_147  cooc_148  cooc_149  \
0    ...        False    False    False    False     False     False   
1    ...        False    False    False    False     False     False   
2    ...        False    False    False    False     False     False   
3    ...        False    False    False    False     False     False   
4    ...        False    False    False    False     False     False   

   cooc_150  cooc_151  class   DateTime  
0     False     False     16 2016-08-08  
1     False     False    133 2016-08-08  
2     False     False     16 2016-08-08  
3     False     False     13 2016-08-08  
4     False     False    133 2016-08-08  

[5 rows x 198 columns]

#### boolean to int

In [265]:
bool_cols=poke.select_dtypes(['bool']).columns

In [266]:
for col in bool_cols:
    poke[col]=poke[col].astype(int)

In [267]:
poke[bool_cols].head(5)

closeToWater  urban  suburban  midurban  rural  gymIn100m  gymIn250m  \
0             0      1         1         1      0          1          1   
1             0      1         1         1      0          0          0   
2             0      0         1         1      0          0          0   
3             1      1         1         1      0          0          0   
4             1      1         1         1      0          0          1   

   gymIn500m  gymIn1000m  gymIn2500m    ...     cooc_142  cooc_143  cooc_144  \
0          1           1           1    ...            0         0         0   
1          1           1           1    ...            0         0         0   
2          1           1           1    ...            0         0         0   
3          1           1           1    ...            0         0         0   
4          1           1           1    ...            0         0         0   

   cooc_145  cooc_146  cooc_147  cooc_148  cooc_149  cooc_150  cooc_151  
0         0         0         0         0         0         0         0  
1         0         0         0         0         0         0         0  
2         0         0         0         0         0         0         0  
3         0         0         0         0         0         0         0  
4         0         0         0         0         0         0         0  

[5 rows x 168 columns]

In [268]:
poke.drop('DateTime',axis=1,inplace=True)
poke.drop('appearedMonth',axis=1,inplace=True)
poke.drop('appearedYear',axis=1,inplace=True)

In [252]:
poke.head(5)

latitude   longitude  appearedTimeOfDay  appearedHour  appearedMinute  \
0  20.525745  -97.460829                  4             5              57   
1  20.523695  -97.461167                  4             5              57   
2  38.903590  -77.199780                  4             5              57   
3  47.665903 -122.312561                  4             5              56   
4  47.666454 -122.311628                  4             5              56   

   appearedDayOfWeek  appearedDay  terrainType  closeToWater  city  ...    \
0                  1            8           14             0    60  ...     
1                  1            8           14             0    60  ...     
2                  1            8           13             0    66  ...     
3                  1            8            0             1    53  ...     
4                  1            8            0             1    53  ...     

   cooc_143  cooc_144  cooc_145  cooc_146  cooc_147  cooc_148  cooc_149  \
0         0         0         0         0         0         0         0   
1         0         0         0         0         0         0         0   
2         0         0         0         0         0         0         0   
3         0         0         0         0         0         0         0   
4         0         0         0         0         0         0         0   

   cooc_150  cooc_151  class  
0         0         0     16  
1         0         0    133  
2         0         0     16  
3         0         0     13  
4         0         0    133  

[5 rows x 195 columns]

#### weather, city, continent to int

In [264]:
le.fit(poke.weather.unique())
list(le.classes_)
poke['weather']=le.transform(poke['weather'])+1

#### global

In [247]:
le.fit(poke.continent.unique())
list(le.classes_)
poke['continent']=le.transform(poke['continent'])+1

In [246]:
le.fit(poke.city.unique())
list(le.classes_)
poke['city']=le.transform(poke['city'])+1

In [253]:
poke.to_csv("/Users/Shuxian/Documents/CSYE7200-Fall2016/predictemall/poke.csv",index=False)

#### Filter: return America only, around 150k rows * 200 columns

In [269]:
poke=poke.loc[poke['continent'] == 'America']

#### check distinct city names

In [270]:
cities=poke.city.unique()

In [271]:
for city in cities:
    city=city.strip()
    city=city.replace("_"," ")
    print(city)

Mexico City
New York
Los Angeles
Chicago
Edmonton
Phoenix
Vancouver
Denver
Toronto
Monterrey
Winnipeg
Montreal
Puerto Rico
Sao Paulo
Halifax
Regina
Guayaquil
Detroit
Bahia
Boise
Bogota
Santiago
Costa Rica
Fortaleza
Belem
Guyana


#### 'New_York', 'Los_Angeles', 'Chicago','Phoenix','Denver','Puerto_Rico','Detroit','Boise

#### US cities only ~120k rows

In [272]:
uscities=['New_York', 'Los_Angeles', 'Chicago','Phoenix','Denver','Puerto_Rico','Detroit','Boise']

In [273]:
poke_us=poke[poke['city'].isin(uscities)]

In [274]:
poke_us.drop('continent',axis=1,inplace=True)

In [276]:
le.fit(poke_us.city.unique())
list(le.classes_)
poke_us['city']=le.transform(poke_us['city'])+1

In [278]:
poke_us.to_csv("/Users/Shuxian/Documents/CSYE7200-Fall2016/predictemall/poke_us.csv",index=False)

geolocator = Nominatim()
location = geolocator.reverse("20.525745, -97.460829")
location.address
location.raw
